docs/covering-dates.bnf

```diff
@@ -0,0 +1,27 @@
+<month> ::= "Jan" | "Feb" | "Mar" | "Apr" | "May" | "June" | "July" | "Aug" | "Sept" | "Oct" | "Nov" | "Dec"
+
+<year> ::= valid year
+
+<year_month> ::= <year> <month>
+
+<year_month_day> ::= <year> <month> valid day of month
+
+<single> ::= <year> | <year_month> | <year_month_day>
+
+<range> ::= <single> "-" <single>
+
+<approx> ::= "?" <range> | "?" <single> | "c" <range> | "c" <single>
+
+<derived> ::= "[" <range> "]" | "[" <single> "]" | "[" <approx> "]"
+
+<undated> ::= "undated"
+
+<single_or_range> ::= <single> | <range>
+
+<gap_repeating> ::= <gap_date> | <single_or_range>
+
+// I assume there can be an unbounded number of "gaps" - the only example I
+// have is "1868; 1890-1902; 1933"
+<gap_date> ::= <single_or_range> ";" <gap_repeating>
+
+<covering-date> ::= <range> | <single> | <derived> | <appox> | <undated> | <gap_date>
```

Comment:
I think we need `Between` in here somewhere as dates such as `Between 1558-1603` are valid according to the spec.

Reply:
The section in the spec about "Between":
```
For example, for certain dates in the reign of Elizabeth I, enter 1558-1603 in Covering Dates in order to generate appropriate start and end dates; then in Covering Dates precede these with ‘Between’ – i.e. Between 1558-1603
```

It _sounds_ like the current system cannot parse "between" expression to start and end dates, and the "between" bit has to be entered manually after the calculation has been done.

It's easy enough to add a parser for it, so I might as well. I'll include it as a seperate commit though.